[Index Management] Mappings empty state

wildemat opened the pull request:
@elastic/kibana-management: I need your input on a display change to index mappings that affects the mapping pages within stack management. 

This is part of a larger effort to redesign and consolidate index management between search and stack management. See [[Epic][Design] Index Management Improvements](https://github.com/elastic/search-team/issues/9365). 

## Summary
When a user has no index mappings, we'd like to show a minimalistic prompt ([design](https://github.com/elastic/search-team/issues/9737)) for field creation:

<kbd><img width="852" height="421" alt="Screenshot 2025-08-19 at 10 57 44 AM" src="https://github.com/user-attachments/assets/6977be6a-b8e6-4389-b475-c93c32d80a8c" /></kbd>


#### There are 4 variations of the mappings tab (screenshots below):
1. Stack Management -> Index Management
2. Stack Management -> Content Connectors
3. Search Solution -> Index Management
4. Search Solution -> Content Connectors

Within search, we'd like to just show the new empty prompt (`Add fields to your data...`) and hide the extra context items (`About index mappings...` and `Transform your searchable content`).

I'd like to know if @elastic/kibana-management would like the same treatment, or if I should update this PR to only apply the empty state to search solution. 

We can also try a variation where the right-hand context items do not display in the empty state, but show up again when fields are added.

The current state of this PR shows the empty state on every mappings tab, alongside whatever extra context items are already there.

There is duplicated code across these various mappings displays. (see #213509 for example). This PR isn't meant to solve that. However I think if we agree on a single version to show, it'll make that task easier.

## Changes
* Leverage `EuiEmptyPrompt` to display a CTA to add a field
* During field creation, if they have no saved fields yet, hide the "filter" bar and "list/json" buttons. Those don't provide helpful context until after fields exist.
* Provide a link to mapping documentation
* Added a few unit tests to verify the proper show/hide behavior of the empty prompt


## Screenshots

### Sample interaction:

https://github.com/user-attachments/assets/03015f96-9b9d-404d-843b-02b5dfb78f4a


### Stack Management Index Management:
#### path: `app/management/data/index_management/indices/index_details?indexName={indexName}&tab=mappings`

<kbd>
<img width="1487" height="815" alt="stack classic mgmt indexmgmt" src="https://github.com/user-attachments/assets/0808495f-4454-4fd9-9b38-92526899d936" />

</kbd>

### Stack Management Content Connectors:
#### path: `app/management/data/content_connectors/connectors/{connectorID}/index_mappings`

<kbd>
<img width="1442" height="852" alt="stack classic mgmt connectors" src="https://github.com/user-attachments/assets/84ac974d-cfaa-45b5-82cd-69ce6ba2b4d0" />

</kbd>

### Search Solution Index Management:
#### path: `search/app/elasticsearch/indices/index_details/{indexName}/mappings`

<kbd>
<img width="1492" height="687" alt="stack search indices" src="https://github.com/user-attachments/assets/07651902-d7e7-4455-b536-f3c2ac11a72e" />

</kbd>

### Search Solution Content Connectors:
#### path: `search/app/elasticsearch/content/connectors/{connectorID}/index_mappings`

<kbd>
<img width="1476" height="789" alt="stack search connectors entsearch" src="https://github.com/user-attachments/assets/90a1aa5a-d19c-45f9-ade5-ce85507aef52" />

</kbd>


## Tests

```bash
➜  kibana git:(search/indices/mappings/empty-state) ✗ yarn test:jest \
'x-pack/platform/plugins/shared/index_management/__jest__/client_integration/index_details_page/index_details_page.test.tsx' \
-t '\<IndexDetailsPage /\> Mappings tab' \
-c './x-pack/platform/plugins/shared/index_management/jest.config.js' \
--verbose=false

yarn run v1.22.22

 PASS  x-pack/platform/plugins/shared/index_management/__jest__/client_integration/index_details_page/index_details_page.test.tsx (14.015 s)

Test Suites: 1 passed, 1 total
Tests:       63 skipped, 25 passed, 88 total
Snapshots:   0 total
Time:        14.332 s, estimated 15 s
Ran all test suites matching /x-pack\/platform\/plugins\/shared\/index_management\/__jest__\/client_integration\/index_details_page\/index_details_page.test.tsx/i with tests matching "\<IndexDetailsPage /\> Mappings tab".
 debg CIStatsReporter committerHash: 85e34139f9a5740b9cd7
✨  Done in 17.15s.
```

Closes https://github.com/elastic/search-team/issues/9737

wildemat committed:
first version empty state for index mappings

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

wildemat committed:
remove comment

wildemat committed:
share the AddFieldButton between with empty mappings prompt. Add tests

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/330867)
* Commit: 0d5a7f30d7c9a960c88bf533777da41e80d29c3d

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/330867#0198c2df-9eca-43b3-8613-d6a7f1c5fc7c)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330867","number":330867,"commit":"0d5a7f30d7c9a960c88bf533777da41e80d29c3d"}],"number":330867}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/330900)
* Commit: 1ff7a1dff0d24f531e7d169ca5edb4287883a6e7

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/68afd7ed09f2053529fc8ec8e4ec859c415ae1f7) | [after](https://github.com/elastic/kibana/commit/1ff7a1dff0d24f531e7d169ca5edb4287883a6e7) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 668 | 669 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/68afd7ed09f2053529fc8ec8e4ec859c415ae1f7) | [after](https://github.com/elastic/kibana/commit/1ff7a1dff0d24f531e7d169ca5edb4287883a6e7) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 681.6KB | 682.9KB | +1.2KB |



### History
* :broken_heart: [Build #330867](https://buildkite.com/elastic/kibana-pull-request/builds/330867) failed 0d5a7f30d7c9a960c88bf533777da41e80d29c3d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330900","number":330900,"commit":"1ff7a1dff0d24f531e7d169ca5edb4287883a6e7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330867","number":330867,"commit":"0d5a7f30d7c9a960c88bf533777da41e80d29c3d"}],"number":330900}
buildkite-pr-comment-->

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

ElenaStoeva reviewed (commented):
Thanks for working on this @wildemat! It completelly makes sense to me to have the same empty state for all index pages in Kibana, but I'll first discuss it with the team to see if everyone is onboard - I added this to our next team sync-up agenda.

Also, I noticed that when Kibana is in High contrast mode, a weird border is being displayed - probably surrounding the component below the empty state promp:

<img width="1163" height="576" alt="Image" src="https://github.com/user-attachments/assets/6aaa0208-c225-4f73-8b6d-531973429e9e" />

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335484)
* Commit: 87e6009675881f2325e49cf7a64d7b9c0c89b816

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/7ffe832b109f5df2e42a655072c24c0c97e07191) | [after](https://github.com/elastic/kibana/commit/87e6009675881f2325e49cf7a64d7b9c0c89b816) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 673 | 674 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7ffe832b109f5df2e42a655072c24c0c97e07191) | [after](https://github.com/elastic/kibana/commit/87e6009675881f2325e49cf7a64d7b9c0c89b816) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 689.5KB | 690.7KB | +1.2KB |



### History
* :yellow_heart: [Build #333251](https://buildkite.com/elastic/kibana-pull-request/builds/333251) was flaky 4757af9772dd7a918f5c44c49e143ac4c3f06104
* :green_heart: [Build #331878](https://buildkite.com/elastic/kibana-pull-request/builds/331878) succeeded 9e742d3a7d26e8958531e4f70f3eca0e4f0659c2
* :green_heart: [Build #330900](https://buildkite.com/elastic/kibana-pull-request/builds/330900) succeeded 1ff7a1dff0d24f531e7d169ca5edb4287883a6e7
* :broken_heart: [Build #330867](https://buildkite.com/elastic/kibana-pull-request/builds/330867) failed 0d5a7f30d7c9a960c88bf533777da41e80d29c3d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335484","number":335484,"commit":"87e6009675881f2325e49cf7a64d7b9c0c89b816"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333251","number":333251,"commit":"4757af9772dd7a918f5c44c49e143ac4c3f06104"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331878","number":331878,"commit":"9e742d3a7d26e8958531e4f70f3eca0e4f0659c2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330900","number":330900,"commit":"1ff7a1dff0d24f531e7d169ca5edb4287883a6e7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330867","number":330867,"commit":"0d5a7f30d7c9a960c88bf533777da41e80d29c3d"}],"number":335484}
buildkite-pr-comment-->

mattkime commented:
It would be nice if the design scaled well for smaller screen sizes

<img width="819" height="740" alt="Screenshot 2025-09-03 at 10 22 45 PM" src="https://github.com/user-attachments/assets/f9c935a2-fd32-4044-a794-c03669976933" />

mattkime commented:
I'm curious why Search Solution Index Management doesn't have the explanatory text. Not looking to debate it, just want to verify its been considered.

mattkime reviewed (changes requested):
There are some relatively minor issues that myself and others have pointed out although it looks like its on the right track overall. I'll take another look when these have been addressed.

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/337529)
* Commit: bd6a236e21e3ee83f91425b298cfcda8c2d33672

### Failed CI Steps
* [FTR Configs #79](https://buildkite.com/elastic/kibana-pull-request/builds/337529#01992fb3-fb8a-4010-b7b7-7dc4683277d9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/337529#01992fb3-fb8a-4010-b7b7-7dc4683277d9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/337529/jobs/01992fb3-fb8a-4010-b7b7-7dc4683277d9/artifacts/01992fda-d0d7-4f85-a86a-003e09a1f036) FTR Configs #<span></span>79 / saved objects tagging - functional tests feature controls tag management write privileges can bulk delete tags

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/19df1a8986d056f9da1c476fc8ad6caf073bed86) | [after](https://github.com/elastic/kibana/commit/bd6a236e21e3ee83f91425b298cfcda8c2d33672) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 673 | 674 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/19df1a8986d056f9da1c476fc8ad6caf073bed86) | [after](https://github.com/elastic/kibana/commit/bd6a236e21e3ee83f91425b298cfcda8c2d33672) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 689.5KB | 690.7KB | +1.2KB |



### History
* :green_heart: [Build #335484](https://buildkite.com/elastic/kibana-pull-request/builds/335484) succeeded 87e6009675881f2325e49cf7a64d7b9c0c89b816
* :yellow_heart: [Build #333251](https://buildkite.com/elastic/kibana-pull-request/builds/333251) was flaky 4757af9772dd7a918f5c44c49e143ac4c3f06104
* :green_heart: [Build #331878](https://buildkite.com/elastic/kibana-pull-request/builds/331878) succeeded 9e742d3a7d26e8958531e4f70f3eca0e4f0659c2
* :green_heart: [Build #330900](https://buildkite.com/elastic/kibana-pull-request/builds/330900) succeeded 1ff7a1dff0d24f531e7d169ca5edb4287883a6e7
* :broken_heart: [Build #330867](https://buildkite.com/elastic/kibana-pull-request/builds/330867) failed 0d5a7f30d7c9a960c88bf533777da41e80d29c3d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337529","number":337529,"commit":"bd6a236e21e3ee83f91425b298cfcda8c2d33672"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335484","number":335484,"commit":"87e6009675881f2325e49cf7a64d7b9c0c89b816"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333251","number":333251,"commit":"4757af9772dd7a918f5c44c49e143ac4c3f06104"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331878","number":331878,"commit":"9e742d3a7d26e8958531e4f70f3eca0e4f0659c2"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330900","number":330900,"commit":"1ff7a1dff0d24f531e7d169ca5edb4287883a6e7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330867","number":330867,"commit":"0d5a7f30d7c9a960c88bf533777da41e80d29c3d"}],"number":337529}
buildkite-pr-comment-->

wildemat committed:
Consolidate index mapping support content. Remove unused extension function

wildemat committed:
move semantic banner up

wildemat committed:
remove debug text

wildemat commented:
### New changes: 
* consolidated "About" and "transform" content to one version
* removed the now unused `setIndexMappingContent` function in the extension.
* removed the manually displayed "About" and "Transform" code from content connectors.
* Fixed empty div showing on high contrast mode

### Screenshots:

#### Empty Mappings
<kbd><img width="1629" height="859" alt="stackIndicesEmpty" src="https://github.com/user-attachments/assets/8c1dc1be-17f0-4ebb-9388-64f1bdf2469d" /></kbd>

#### Non-Empty Mappings
<kbd><img width="1639" height="771" alt="stackIndicesNonEmpty" src="https://github.com/user-attachments/assets/8db125c7-c207-4d23-b5cf-83f6cc81e230" /></kbd>


https://github.com/user-attachments/assets/06bfea1e-ccf5-4ee9-9734-37618d100e56



### Summary

I attempted the conditional display of the extra help content [here](https://github.com/wildemat/kibana/pull/3), but in the current plugin's architecture it may be too much scope to make that happen for now.

As a compromise, the various versions of "About" and "Transform" content was consolidated into a single version in the index management mapping component.

2 of the alternate versions were from Search Content Connectors and Stack Mgmt Content Connectors. Both of which are being deprecated.

Serverless Search had a different version as well, but it was never being displayed due to `showAboutMappings` set to `false`

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

wildemat committed:
Fix formatting

wildemat commented:
On small screen:
<kbd><img width="1016" height="882" alt="Screenshot 2025-09-17 at 6 01 17 PM" src="https://github.com/user-attachments/assets/037d889a-f562-4493-8156-454a86751807" /></kbd>

wildemat committed:
remove redundant flex props

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340579)
* Commit: 7385ca2181aecc31a684e52d245b18b6727ea3db

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/340579#019959b7-6e0e-4084-80ba-5a99697d2c66)

### History
* :broken_heart: [Build #340567](https://buildkite.com/elastic/kibana-pull-request/builds/340567) failed f3e44fceefbbc247266cc4ad4cb4a792eceac47d
* :yellow_heart: [Build #337529](https://buildkite.com/elastic/kibana-pull-request/builds/337529) was flaky bd6a236e21e3ee83f91425b298cfcda8c2d33672
* :green_heart: [Build #335484](https://buildkite.com/elastic/kibana-pull-request/builds/335484) succeeded 87e6009675881f2325e49cf7a64d7b9c0c89b816
* :yellow_heart: [Build #333251](https://buildkite.com/elastic/kibana-pull-request/builds/333251) was flaky 4757af9772dd7a918f5c44c49e143ac4c3f06104
* :green_heart: [Build #331878](https://buildkite.com/elastic/kibana-pull-request/builds/331878) succeeded 9e742d3a7d26e8958531e4f70f3eca0e4f0659c2

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340579","number":340579,"commit":"7385ca2181aecc31a684e52d245b18b6727ea3db"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340567","number":340567,"commit":"f3e44fceefbbc247266cc4ad4cb4a792eceac47d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337529","number":337529,"commit":"bd6a236e21e3ee83f91425b298cfcda8c2d33672"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335484","number":335484,"commit":"87e6009675881f2325e49cf7a64d7b9c0c89b816"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333251","number":333251,"commit":"4757af9772dd7a918f5c44c49e143ac4c3f06104"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331878","number":331878,"commit":"9e742d3a7d26e8958531e4f70f3eca0e4f0659c2"}],"number":340579}
buildkite-pr-comment-->

wildemat committed:
Fix i18n issues with 'node scripts/i18n_check --fix'

wildemat commented:
Note, the following data-test-subj items will be removed:
- Connectors in stack and enterprise search:
	- enterpriseSearchSearchIndexIndexMappingsLearnHowToCustomizeIndexMappingsAndSettingsLink
	- enterpriseSearchSearchIndexIndexMappingsLearnMoreLink
- Search indices
	- serverlessSearchIndexMappingsDocsLinkLearnMoreAboutIngestPipelinesLink
	
They will be replaced with 
	- indexDetailsMappingsDocsLink
	- indexDetailsMappingsLearnMoreLink

wildemat committed:
type fix

wildemat committed:
remove references to index mappings content in extension service

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/341071)
* Commit: f33c9cace7a66f82c21b850e63e2e00c964ee7ee
* [Cloud Deployment](https://kibana-pr-231899-28cb57.kb.us-west2.gcp.elastic-cloud.com)

### History
* :yellow_heart: [Build #340933](https://buildkite.com/elastic/kibana-pull-request/builds/340933) was flaky c5893eccdfbe985981b52d09601960308e9ebb1b
* :broken_heart: [Build #340789](https://buildkite.com/elastic/kibana-pull-request/builds/340789) failed 0bb940e1a43d37394a8296b393e8598c56c4a2bb
* :broken_heart: [Build #340582](https://buildkite.com/elastic/kibana-pull-request/builds/340582) failed 63c9db148c01064455863aca6b5f50b653c4a135
* :broken_heart: [Build #340579](https://buildkite.com/elastic/kibana-pull-request/builds/340579) failed 7385ca2181aecc31a684e52d245b18b6727ea3db
* :broken_heart: [Build #340567](https://buildkite.com/elastic/kibana-pull-request/builds/340567) failed f3e44fceefbbc247266cc4ad4cb4a792eceac47d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341071","number":341071,"commit":"f33c9cace7a66f82c21b850e63e2e00c964ee7ee"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340933","number":340933,"commit":"c5893eccdfbe985981b52d09601960308e9ebb1b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340789","number":340789,"commit":"0bb940e1a43d37394a8296b393e8598c56c4a2bb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340582","number":340582,"commit":"63c9db148c01064455863aca6b5f50b653c4a135"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340579","number":340579,"commit":"7385ca2181aecc31a684e52d245b18b6727ea3db"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340567","number":340567,"commit":"f3e44fceefbbc247266cc4ad4cb4a792eceac47d"}],"number":341071}
buildkite-pr-comment-->

seialkali reviewed (approved):
LGTM

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

wildemat committed:
fixup

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/342226)
* Commit: 00033249c73cff1dd68619020e340c686e6b3611

### Failed CI Steps
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/342226#0199728f-8d67-4ab1-8308-bd6fe39ac3cb)
* [Build and Deploy to Cloud](https://buildkite.com/elastic/kibana-pull-request/builds/342226#0199726b-ac1a-443e-87e7-eb3b8cc52ea6)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342226#0199728f-8d67-4ab1-8308-bd6fe39ac3cb) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342226/jobs/0199728f-8d67-4ab1-8308-bd6fe39ac3cb/artifacts/019972a6-b928-42f1-a97f-c13694e4e1cc) Scout: [ platform / streams_app ] plugin / stateful - Stream data routing - reordering routing rules - should handle multiple reorder operations

### History
* :broken_heart: [Build #342194](https://buildkite.com/elastic/kibana-pull-request/builds/342194) failed 0d84009c5ba2d83352023d3d2df0dc136fb3a00d
* :yellow_heart: [Build #341071](https://buildkite.com/elastic/kibana-pull-request/builds/341071) was flaky f33c9cace7a66f82c21b850e63e2e00c964ee7ee
* :yellow_heart: [Build #340933](https://buildkite.com/elastic/kibana-pull-request/builds/340933) was flaky c5893eccdfbe985981b52d09601960308e9ebb1b
* :broken_heart: [Build #340789](https://buildkite.com/elastic/kibana-pull-request/builds/340789) failed 0bb940e1a43d37394a8296b393e8598c56c4a2bb
* :broken_heart: [Build #340582](https://buildkite.com/elastic/kibana-pull-request/builds/340582) failed 63c9db148c01064455863aca6b5f50b653c4a135

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342226","number":342226,"commit":"00033249c73cff1dd68619020e340c686e6b3611"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342194","number":342194,"commit":"0d84009c5ba2d83352023d3d2df0dc136fb3a00d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/341071","number":341071,"commit":"f33c9cace7a66f82c21b850e63e2e00c964ee7ee"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340933","number":340933,"commit":"c5893eccdfbe985981b52d09601960308e9ebb1b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340789","number":340789,"commit":"0bb940e1a43d37394a8296b393e8598c56c4a2bb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340582","number":340582,"commit":"63c9db148c01064455863aca6b5f50b653c4a135"}],"number":342226}
buildkite-pr-comment-->

wildemat committed:
Merge branch 'main' into search/indices/mappings/empty-state

mattkime reviewed (approved):
Looks good and works well in stack management/ index management
